scripts/extract-ios-device-names.ts

```diff
@@ -81,21 +85,27 @@ const formatOutput = async (unformatted: string) => {
 };
 
 export async function extractIOSDeviceNames() {
-  try {
-    if (fs.statSync(outputPath)) {
-      // Out with the old, in with the new
-      fs.unlinkSync(outputPath);
-    }
-  } catch (e) {
-    // File does not exists, carry along
-  }
-
   const files = await getDefinitionFiles();
   const definitions = await collectDefinitions(files);
   const formatted = await formatOutput(
     template(JSON.stringify(definitions, undefined, 2))
   );
 
-  fs.writeFileSync(outputPath, formatted);
-  console.log('✅ Regenerated ios-device-list.tsx');
+  // All exit code has to synchronous
+  const cleanup = () => {
+    if (fs.existsSync(tmpOutputPath)) {
+      fs.unlinkSync(tmpOutputPath);
+    }
+  };
+
+  process.on('exit', cleanup);
+  process.on('SIGINT', () => {
```

Comment:
I tested interrupting manually by adding a sleep between writesync and rename and the file was removed as expected